Initial code checkin

srp3rd opened the pull request.

srp3rd committed:
Initial checkin of new code

srp3rd committed:
Add workflows

srp3rd committed:
Remove print statements

srp3rd committed:
Update to latest DM stack
Fix issue in Message where str was returned instead of dict
Fix issue in ingestd where dict was requested instead of str
Update log messages to indicate why things might have been configured incorrectly
Remove version.py, since it shouldn't have ever been checked in

srp3rd committed:
Fix so ingest uses "auto" instead of "direct"

srp3rd committed:
Update formatting and lint checkers to latest versions

srp3rd committed:
Fix sidecar returns which caused unittest to fail

srp3rd committed:
Use dict instead of str
fix yaml entry for XRD2

srp3rd committed:
Add more content to README.md
Fix versioning in build

srp3rd committed:
Rename env file

srp3rd committed:
Change "ingestd.env" to "versions.env" on build

srp3rd committed:
Fix to use the CTRL_INGESTD_VERSION

srp3rd committed:
Add missing quote

ktlim commented on README.md:
```diff
@@ -1,2 +1,44 @@
+
+"brokers" is set to the host Kafka "kafka", listening to port 9092.
+
+"group_id" - Kafka group id, which can be set to what you wish, but should be unique if there are any other Kafka clients running.  In this case, "my_test_group".
```
The `group_id` can and actually must be the same if we need to deploy multiple parallel ingesters for performance.

srp3rd replied:
Additional text added to reflect this.

ktlim commented on README.md:
```diff
@@ -1,2 +1,44 @@
+
+group_id: "my_test_group"
+
+num_messages: 50
```
This is not explained in the document.  It's the maximum message batch size processed at one time, I believe.

srp3rd replied:
Added explanatory text

ktlim commented on README.md:
```diff
@@ -1,2 +1,44 @@
+    repo: /tmp/repo
+
+rses:
+    XRD1:
```
It can sometimes be a bit awkward to have YAML keys that are user-provided.  It makes it difficult to validate the files, and the syntax can sometimes be tricky if there are special characters in the key name.

The alternate way of doing this would be to have `rses` be a list rather than a mapping, with each list element being a mapping that contains an element like `name: XRD1` in addition to `rucio_prefix:` and `fs_prefix:`.

srp3rd replied:
I went back on forth on this when I originally did it.  I didn't want to list the RSE names in two places because it was just another place to make a mistake.   Having said that, I can change it if you prefer.

ktlim commented on bin.src/ingestd:
```diff
@@ -0,0 +1,41 @@
+import asyncio
+import logging
+import os
+import sys
```
Only `logging` is actually used now.

srp3rd replied:
Fixed

srp3rd committed:
Update to weekly 19 of the LSST stack

srp3rd committed:
Update version number

ktlim commented on .github/workflows/build.yaml:
```diff
@@ -0,0 +1,37 @@
+      contents: read
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
```
Now at v4...

srp3rd replied:
Fixed

ktlim commented on docker/ingestd/Dockerfile:
```diff
@@ -0,0 +1,24 @@
+
+ARG CONFLUENT_KAFKA_VERSION
+RUN source /opt/lsst/software/stack/loadLSST.bash && \
+    pip install confluent_kafka==${CONFLUENT_KAFKA_VERSION}
```
`mamba install python-confluent-kafka==${CONFLUENT_KAFKA_VERSION}` would be better.

srp3rd replied:
Fixed

ktlim commented on docker/ingestd/Dockerfile:
```diff
@@ -0,0 +1,24 @@
+WORKDIR /home/lsst
+
+ARG CTRL_INGESTD_GIT_VERSION
+RUN git clone https://github.com/lsst-dm/ctrl_ingestd && \ 
```
Better to `eups distrib install` this if possible.  Then the whole `scons` command can go away.

srp3rd replied:
where are the instructions to get this to work via `eups distrib install`?

srp3rd replied:
ctrl_ingestd isn't part of distrib, so we can't do this.

ktlim commented on docker/ingestd/run.sh:
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+source /opt/lsst/software/stack/loadLSST.bash
+setup -r /home/lsst/ctrl_ingestd
```
If this is installed with `eups distrib install` instead of `git clone`, then I think this can be just `setup ctrl_ingestd`.  (Possibly you'll need to make sure the `current` tag is applied in eups.)

srp3rd replied:
ctrl_ingestd isn't part of distrib, so we can't do this.

ktlim commented on docker/testenv-ingestd/Dockerfile:
```diff
@@ -0,0 +1,16 @@
+FROM lsstsqre/centos:${LSST_STACK_VERSION}
+
+USER root
+RUN yum install -y nc
```
Can you add a comment explaining why this is needed?

srp3rd replied:
This install of "nc" should have been removed.   I was using it to debug something early on, and never took it out.   I'll remove it.

ktlim reviewed (commented):
Stumbled back across this after a long time.  I still haven't gotten to the code itself, but I have some comments on the build/deploy portions.

srp3rd committed:
Update to v4

srp3rd committed:
Added addtional explanatory text

srp3rd committed:
Remove unnecessary imports
Change to use mamba instead of pip to install
Remove nc (no longer used)
Update version

srp3rd committed:
Update weekly stack version

srp3rd committed:
Fix syntax error in build.yaml

srp3rd committed:
Revert to weekly 22

srp3rd committed:
Update version number of ctrl_ingestd
